x/smart/stream_dialer.go

```diff
@@ -250,21 +250,31 @@ func (f *StrategyFinder) testDialer(ctx context.Context, dialer transport.Stream
 		testAddr := net.JoinHostPort(testDomain, "443")
 		f.logCtx(ctx, "🏃 running test: '%v' (domain: %v)\n", transportCfg, testDomain)
 
-		ctx, cancel := context.WithTimeout(ctx, f.TestTimeout)
+		testCtx, cancel := context.WithTimeout(ctx, f.TestTimeout)
 		defer cancel()
-		testConn, err := dialer.DialStream(ctx, testAddr)
+
+		timeoutChan := make(chan struct{})
+		go func() {
+			<-testCtx.Done()
+			if errors.Is(testCtx.Err(), context.DeadlineExceeded) {
+				f.logCtx(testCtx, "⏱️ dialer failure, test timed out: '%v' (domain: %v), duration=%v ❌\n", transportCfg, testDomain, time.Since(startTime))
+				close(timeoutChan)
+			}
+		}()
+
+		testConn, err := dialer.DialStream(testCtx, testAddr)
 		if err != nil {
-			f.logCtx(ctx, "🏁 failed to dial: '%v' (domain: %v), duration=%v, dial_error=%v ❌\n", transportCfg, testDomain, time.Since(startTime), err)
+			f.logCtx(testCtx, "🏁 failed to dial: '%v' (domain: %v), duration=%v, dial_error=%v ❌\n", transportCfg, testDomain, time.Since(startTime), err)
```

Comment:
Check the deadline here instead. The DialStream is supposed to return on timeouts.

Reply:
I did this, but at :266 where the timeout is returned, not here.